DependencyInjection/SmartboxIntegrationFrameworkExtension.php

```diff
@@ -271,6 +274,55 @@ protected function loadQueueDrivers(ContainerBuilder $container)
                     }
                     break;
 
+                case 'phpamqplib':
+                    if (!class_exists(AMQPStreamConnection::class)) {
+                        throw new \RuntimeException('You need the amqp extension to use PHP-AMQP-LIB driver.');
+                    }
+
+                    if (empty($driverConfig['connections'] ?? [])) {
+                        throw new \InvalidArgumentException('You need to define at least one connection to use the AMQP driver.');
+                    }
+
+                    $driverDef = new Definition(PhpAmqpLibDriver::class);
+                    $driverDef->addMethodCall('setSerializer', [new Reference('jms_serializer')]);
+                    $driverDef->addMethodCall('setMessageFactory', [new Reference('smartesb.message_factory')]);
```

Comment:
These 2 objects can be passed directly in the service definition. In fact you are already doing that (see Resources/config/services.yml). For this reason this code
```
$driverDef->addMethodCall('setSerializer', [new Reference('jms_serializer')]);
$driverDef->addMethodCall('setMessageFactory', [new Reference('smartesb.message_factory')]);
```
can be deleted

Reply:
These lines were removed.